### ISSUE TITLE
Https Schema Is not Detect with Unix Socket Binding

### ISSUE THREAD

**Issue by myronww** [benoitc/gunicorn#1926]
I am unable to get https schema to pass through using proxy headers.  I added some logging to message.py line 102 to dump out header values.   My nginx server is setup correctly to pass the headers along and the logs are dumping the headers.

Nov 23 20:59:29 thinkbox python[2363]: CRITICAL:werkzeug:MapAdapter.bind: server_name
Nov 23 20:59:29 thinkbox python[2363]: CRITICAL:root:Header: name='HOST' value=['127.0.0.1\r\n']
Nov 23 20:59:29 thinkbox python[2363]: CRITICAL:root:Header: name='X-REAL-IP' value=['127.0.0.1\r\n']
Nov 23 20:59:29 thinkbox python[2363]: CRITICAL:root:Header: name='X-FORWARDED-FOR' value=['127.0.0.1\r\n']
Nov 23 20:59:29 thinkbox python[2363]: CRITICAL:root:Header: name='X-FORWARDED-PROTO' value=['https\r\n']
Nov 23 20:59:29 thinkbox python[2363]: CRITICAL:root:Header: name='X-FORWARDED-SSL' value=['on\r\n']
Nov 23 20:59:29 thinkbox python[2363]: CRITICAL:root:Header: name='X-FORWARDED-PROTOCOL' value=['ssl\r\n']
Nov 23 20:59:29 thinkbox python[2363]: CRITICAL:root:Header: name='USER-AGENT' value=['Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, lik

From looking at the logs, it looks like the issue is that the header parsing code is leaving on line endings and so is propably just failing the schema checks for https because of the "/r/n" on the end of the header values.

**Comment by myronww** [comment]
from looking at the code, value is only being left stripped 

name, value = name.strip(), [value.lstrip()]
 
The schema settings for https do not have "/r/n" on them.

 default = {
        "X-FORWARDED-PROTOCOL": "ssl",
        "X-FORWARDED-PROTO": "https",
        "X-FORWARDED-SSL": "on"
    }

**Comment by myronww** [comment]
Moved my logging further down and was getting the correct headers.  Then I debugged it some and found that remote_addr up at line 78 is of type bytes.  So the following check fails around line 85 and so no secure_scheme_headers are ever set

Change this to:
     elif isinstance(remote_addr, string_types):

To 
      elif isinstance(remote_addr, string_types) or isinstance(remote_addr, binary_type):

And the headers are set.  There is still one mre issue though.  Later on the check below will likely fail because values is a list and not just a string.

secure = value == secure_scheme_headers[name]

The config just has strings so a value that is a list is not likely to equal the string

So I change that line to:
   
secure =  secure_scheme_headers[name] in value

And all is well, works great.

I'll submit a PR
